docs/frontend/architecture/app-structure.md

```diff
@@ -0,0 +1,258 @@
+
+**Advantages of VueMS:**
+
+* Each module can have its **own business context** communicating separately with the backend application.
+* Modules can be **turned active or inactivate** anytime which allows to customize the application for each client individually.
+* Modules can be **loaded locally** within the application and downloaded from external servers (npm, Verdaccio).
+* Thanks to having modules on external servers, we can **version them**, therefore the modules are independent and easy to expand.
+* The customer can **upgrade any package** or **replace it** with another one.
+* Our solution is based on the built-in mechanism of [**NuxtJS modules**][nuxt-modules] in order **not to modify the basic operation of the framework**.
+* The mechanism is **closely related** to [**Vue.js**][vue] and was created only for this framework.
+* The main task of the library is to **transform the monolithic approach** to Nuxt architecture into a modular one.
+* We needed each module to have a **structure similar to a large application**, so that its directory layout <br>
+and mechanisms would not differ from a large monolithic application.
+
+<p align="center">
+    <img src="images/architecture.png"
+        alt="Micro frontend architecture"
+        />
+</p>
+
+
+## Structure
+---
+
+Ergonode in the **developer version** contains in its structure the most important configurations needed for proper operation of the application.
+
+### Configuration files:
+
+* #### Main:
+    * **package.json**<br>
+        Main configuration file for the application based on the npm dependency.<br>
+        We store all information about the project and dependencies in it.
+        > In addition to the file dependency, the file also provides the configuration of the available and required modules. <br>
+        **Only used by CLI.**<br><br>
+        `_requiredModules` - specifies required modules, no module in the application returns error, <br>
+        `_availableModules` - all available modules, local and from the npm server,
+    * **nuxt.config.js** <br>
+        Main [nuxt configuration][nuxt-conf] file.
+        > In here, in addition to the typical configuration, we will set up the [VueMS library configurations][vuems-conf].
+    * **modules.config.js**<br>
+        File created from the CLI level, set configuration for VueMS library - more [here][doc-mod-structure].
+        > CLI command `npm run modules` and  `npm run modules:all` created this file.<br>
+        **File property:**<br>
+        `local` - contains names of local modules,<br>
+        `npm` - contains names of modules hosted on [npm server][npm],
+    * **router.js** <br>
+        [Vue router][vue-router] configuration file.
+        > The default file is extended with additional configurations from the [VueMS library][vuems]
+    * **router.local.js** <br>
+        Local routing file if needed.
+        > Better idea is use modules and there set routing or extend existing routing.
```

Comment:
A better idea would be to use modules and set routing there or extend the existing routing.